modules/libcom/src/osi/os/WIN32/osdFindSymbol.c

```diff
@@ -28,19 +47,46 @@ epicsShareFunc void * epicsLoadLibrary(const char *name)
 
 epicsShareFunc const char *epicsLoadError(void)
 {
-    static char buffer[100];
+    STORE char buffer[100];
+    DWORD n;
 
-    FormatMessage(
+    n = FormatMessage(
         FORMAT_MESSAGE_FROM_SYSTEM,
         NULL,
         epicsLoadErrorCode,
         0,
         buffer,
         sizeof(buffer)-1, NULL );
+
+    /* n - number of chars stored excluding nil.
+     *
+     * Some messages include a trailing newline, which we strip.
+     */
+    for(; n>=1 && (buffer[n-1]=='\n' || buffer[n-1]=='\r'); n--)
+        buffer[n-1] = '\0';
+
     return buffer;
 }
 
 epicsShareFunc void * epicsShareAPI epicsFindSymbol(const char *name)
 {
-    return GetProcAddress(0, name);
+    HMODULE dlls[128];
+    DWORD ndlls=0u, i;
+    void* ret = NULL;
+
+    /* As a handle returned by LoadLibrary() isn't available to us,
+     * try all loaded modules in arbitrary order.
+     */
+    if(epicsEnumProcessModules(GetCurrentProcess(), dlls, sizeof(dlls), &ndlls)) {
+        for(i=0; !ret && i<ndlls; i++) {
+            ret = GetProcAddress(dlls[i], name);
+        }
+    }
```

Comment:
The return argument from epicsEnumProcessModules() is not the number of DLLs but the number of bytes that would be required to store them, you would need to divide by sizeof(HMODULE) to get the real number of DLLs. It could also potentially return a number larger number than passed, so you would need to be compared to sizeof(dlls) and if it is bigger than this number reallocate dlls to a larger size and call again. It may thus be better to call initially with a size of zero and then malloc dlls based on the returned lpcbNeeded

Reply:
yes... oops.